### PR TITLE
CSE-262 define model for app

### PR DESCRIPTION
**Pull request by danvanhorn**
# Changes introduced

I added the data model types from @Cleverus3rname [from his github repo](https://github.com/Cleverus3rname/SB-CSE-Data-Structure/blob/master/README.md). 

## Related issue(s):

- CSE-262

## Contributor checklist

- [ ] ~~Wrote/updated tests for introduced changes~~
- [ ] ~~Added new stories for created/modified components (if applicable)~~
- [ ] ~~Updated Postman library for created/modified routes (if applicable)~~
- [x] Verified that there are no issues in CircleCI
- [x] Assigned yourself to the PR
- [x] Removed `WIP:` from the PR title
- [x] Requested review from the SB Reviewers team

## Reviewer checklist

- [x] Assigned yourself to the PR
- [ ] Read through the changes in the `Files changed` tab
- [ ] Verified that tests were written/modified
- [ ] Verified that stories were written/modified (if applicable)
- [ ] ~~Verified that Postman was updated (if applicable)~~
- [ ] Checked out the branch and tested changes locally

### CODE REVIEW

**Comment by puregarlic** [comment]
Waiting on a data model change.

**Comment by danvanhorn** [comment]
@osu-cass/sb-reviewers  I've updated the data model from the latest example in Evan's repo.

File: .gitignore [hunk 1]
```diff
@@ -2,7 +2,4 @@ node_modules
 dist
 coverage
 reports
-
-# IntelliJ Idea
-.idea/
-*.iml
+settings.json
```
Comment (puregarlic) [inline .gitignore]: We should keep the IntelliJ Idea ignores, since we have a developer that uses that editor.

**Review by puregarlic — CHANGES_REQUESTED** [review]
Add the IntelliJ Idea ignores back to the `.gitignore`, and then this should be fine.

File: .gitignore (hunk 1 above)
Comment (danvanhorn) [reply .gitignore]: I re-added the intellij stuff

**Review by puregarlic — APPROVED** [review]

**Pull request merged**